modules/geo-layers/src/h3-layers/h3-hexagon-layer.js

```diff
@@ -22,9 +23,27 @@ function getHexagonCentroid(getHexagon, object, objectInfo) {
   return [lng, lat];
 }
 
-function h3ToPolygon(hexId) {
+export function scalePolygon(hexId, vertices, factor) {
+  const [lat, lng] = h3ToGeo(hexId);
+  const vertexCount = vertices.length;
+  log.assert(vertices[0] === vertices[vertexCount - 1]);
+  // `h3ToGeoBoundary` returns same array object for first and last vertex (closed polygon),
+  // hence skip scaling the last vertex
+  for (let i = 0; i < vertexCount - 1; i++) {
+    vertices[i][0] = lerp(lng, vertices[i][0], factor);
+    vertices[i][1] = lerp(lat, vertices[i][1], factor);
+  }
+}
+
+function h3ToPolygon(hexId, coverage = 1) {
   const vertices = h3ToGeoBoundary(hexId, true);
   const refLng = vertices[0][0];
+
+  if (coverage !== 1) {
+    // updates array object elements of vertices array.
+    scalePolygon(hexId, vertices, coverage);
```

Comment:
This breaks if the hexagon crosses the 180th meridian. You need to set the refLng to that of the center, normalize it before scaling.

Reply:
Done.